<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>SuRE-SNP project, Figure panels for paper: overlap with eQTLs</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>





<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<p><html></p>

<p><head></p>

<style type="text/css">
.knitr .inline {
  background-color: #f7f7f7;
  border:solid 1px #B0B0B0;
}
.error {
    font-weight: bold;
    color: #FF0000;
}
.warning {
    font-weight: bold;
}
.message {
    font-style: italic;
}
.source, .output, .warning, .error, .message {
    padding: 0 1em;
  border:solid 1px #F7F7F7;
}
.source {
  background-color: #f5f5f5;
}
.rimage .left {
  text-align: left;
}
.rimage .right {
  text-align: right;
}
.rimage .center {
  text-align: center;
}
.hl.num {
  color: #AF0F91;
}
.hl.str {
  color: #317ECC;
}
.hl.com {
  color: #AD95AF;
  font-style: italic;
}
.hl.opt {
  color: #000000;
}
.hl.std {
  color: #585858;
}
.hl.kwa {
  color: #295F94;
  font-weight: bold;
}
.hl.kwb {
  color: #B05A65;
}
.hl.kwc {
  color: #55aa55;
}
.hl.kwd {
  color: #BC5A65;
  font-weight: bold;
}
</style>

<p><title>knitr document van Steensel lab</title></p>

<style type="text/css">
body {font-family:"Helvetica", sans-serif;}
h3 {color: #335599;  margin-bottom: 0em;margin-top: 1em;}
p {margin-top: 0em;}
</style>

<p></head></p>

<p><body></p>

<h1>SuRE-SNP project, Figure panels for paper: overlap with eQTLs</h1>

<h2>Bas van Steensel, May 2018, updated Sept 2018. </h2>

<h3>Introduction</h3>

<p>Some examples of &ldquo;simple&rdquo; and &ldquo;complex&rdquo; cases of SuRe-eQTL overlap, formatted as figure panels.</p>

<h3>Version history</h3>

<ul>
<li>Code chunks and snippets taken from SuRE_SNP_fullset_eQTLoverlap_anecdotes_K562.html</li>
<li>Starting from FigurePanels_SuRE_SNP_eQTLoverlap_BvS180924.Rhtml: SuRE now plotted in lin space instead of log space </li>
<li>Starting from 180927: integrated with Gviz plotting of annotation track and optional DNaseI track for K562</li>
</ul>

<h3>Description of Data</h3>

<p><ul>
<li>SuRE data are in a dataframe generated by Joris. This is the full set of SNPs, NO multiple testing correction by IHW.  </li>
<li>Gtex eQTL significant gene/SNP pair list from all tissues. Downloaded from https://gtexportal.org/home/datasets on 27 Jan 2018, version v7. </li>
<li>K562 DNaseI track is from John Stam lab, ENCODE project, via UCSC genome browser portal.</li>
<li>TF ChIP-seq peak positions (not used here) are from Remap-2018 </li>
</ul>
</p>

<h3>Path, Libraries, Parameters and Functions</h3>

<div class="chunk" id="html-Path&Pars"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl com">#directory:</span>
<span class="hl kwd">setwd</span><span class="hl std">(</span><span class="hl str">&quot;~/mydata/Projects/Git/SuRE/Bas/SNP/eQTL&quot;</span><span class="hl std">)</span>

<span class="hl com">#general stuff:</span>
<span class="hl std">StartTime</span><span class="hl kwb">&lt;-</span><span class="hl kwd">Sys.time</span><span class="hl std">()</span> <span class="hl com">#used to determine runtime, see very end of this doc</span>
<span class="hl std">Date</span><span class="hl kwb">&lt;-</span><span class="hl kwd">substr</span><span class="hl std">(</span><span class="hl kwd">gsub</span><span class="hl std">(</span><span class="hl str">&quot;-&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwd">Sys.time</span><span class="hl std">()),</span><span class="hl num">3</span><span class="hl std">,</span><span class="hl num">8</span><span class="hl std">)</span>


<span class="hl com">#libraries:</span>
<span class="hl com">#for installing uncomment these:</span>
<span class="hl com">#source(&quot;https://bioconductor.org/biocLite.R&quot;)</span>
<span class="hl com">#biocLite(&quot;Gviz&quot;) </span>
<span class="hl kwd">library</span><span class="hl std">(rtracklayer)</span>
<span class="hl kwd">library</span><span class="hl std">(Gviz)</span>
<span class="hl kwd">library</span><span class="hl std">(data.table)</span> <span class="hl com">#faster loading of large tab-delimited files</span>

<span class="hl com">#external data URLs:</span>
<span class="hl std">URL.TF.K562</span><span class="hl kwb">&lt;-</span><span class="hl str">&quot;http://pedagogix-tagc.univ-mrs.fr/remap/download/remap2018/hg19/MACS/cell_line/K562/remap2018_K562_all_macs2_hg19_v1_2.bed.gz&quot;</span>
<span class="hl std">URL.DNase.K562</span><span class="hl kwb">&lt;-</span><span class="hl str">&quot;http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwDgf/wgEncodeUwDgfK562Raw.bigWig&quot;</span>
<span class="hl std">URL.DNase.HepG2</span><span class="hl kwb">&lt;-</span><span class="hl str">&quot;http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeUwDnase/wgEncodeUwDnaseHepg2RawRep1.bigWig&quot;</span>

<span class="hl com">#parameters:</span>
<span class="hl std">p2width</span><span class="hl kwb">&lt;-</span><span class="hl num">1.5</span> <span class="hl com">#factor for translating  -log10(p.value) into line width in plots</span>

<span class="hl com">#plot colors:</span>
<span class="hl std">COLc</span><span class="hl kwb">&lt;-</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;#&quot;</span><span class="hl std">,</span> <span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl kwd">as.hexmode</span><span class="hl std">(</span><span class="hl kwd">col2rgb</span><span class="hl std">(</span><span class="hl str">&quot;darkorange2&quot;</span><span class="hl std">)),</span> <span class="hl kwc">collapse</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">),</span> <span class="hl str">&quot;aa&quot;</span><span class="hl std">)</span> <span class="hl com">#concordant color</span>
<span class="hl std">COLd</span><span class="hl kwb">&lt;-</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;#&quot;</span><span class="hl std">,</span> <span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl kwd">as.hexmode</span><span class="hl std">(</span><span class="hl kwd">col2rgb</span><span class="hl std">(</span><span class="hl str">&quot;royalblue1&quot;</span><span class="hl std">)),</span> <span class="hl kwc">collapse</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">),</span> <span class="hl str">&quot;aa&quot;</span><span class="hl std">)</span> <span class="hl com">#discordant color</span>
</pre></div>

<p></div></div></p>

<h3>Load & preprocess data</h3>

<p>Load SuRE data and SNP2TFBS data, do some basic processing:</p>

<div class="chunk" id="html-LoadProcessData"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl com">#load Joris' complete SuRE dataset:</span>
<span class="hl std">D</span><span class="hl kwb">&lt;-</span><span class="hl kwd">readRDS</span><span class="hl std">(</span><span class="hl str">&quot;/DATA/usr/joris/git/SuRE_shared/Joris/SNP_project/all_snp_analysis_wilcox/covered.10.1000.sure.snp.df.strand.nonspecific.exact.wilcox.downsampled.JvA.180828.rds&quot;</span><span class="hl std">)</span>
<span class="hl com">#construct chrpos in D, for linking to eQTLs:</span>
<span class="hl std">D</span><span class="hl opt">$</span><span class="hl std">chrpos</span><span class="hl kwb">&lt;-</span><span class="hl kwd">paste</span><span class="hl std">(</span><span class="hl kwd">gsub</span><span class="hl std">(</span><span class="hl str">&quot;chr&quot;</span><span class="hl std">,</span><span class="hl str">&quot;&quot;</span><span class="hl std">,D</span><span class="hl opt">$</span><span class="hl std">chr), D</span><span class="hl opt">$</span><span class="hl std">SNPabspos,</span> <span class="hl kwc">sep</span><span class="hl std">=</span><span class="hl str">'_'</span><span class="hl std">)</span> <span class="hl com">#this we are going to use to link databases</span>
<span class="hl kwd">dim</span><span class="hl std">(D)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 5919239      26
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#load Joris' list of signifant SuRE hits K562:</span>
<span class="hl std">DhitsK</span><span class="hl kwb">&lt;-</span><span class="hl kwd">readRDS</span><span class="hl std">(</span><span class="hl str">&quot;/DATA/usr/joris/git/SuRE_shared/Joris/SNP_project/180813_fdr_analysis/k562.sign.id.JvA180828.rds&quot;</span><span class="hl std">)</span>
<span class="hl kwd">length</span><span class="hl std">(DhitsK)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 19237
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#load Joris' list of expression-matched set of non-hits K562:</span>
<span class="hl std">DmatchK</span><span class="hl kwb">&lt;-</span><span class="hl kwd">readRDS</span><span class="hl std">(</span><span class="hl str">&quot;/DATA/usr/joris/git/SuRE_shared/Joris/SNP_project/180910_Matched_SNPs/k562.matched.control.180910.rds&quot;</span><span class="hl std">)</span>
<span class="hl kwd">length</span><span class="hl std">(DmatchK)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 19237
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#load Joris' list of signifant SuRE hits HepG2:</span>
<span class="hl std">DhitsH</span><span class="hl kwb">&lt;-</span><span class="hl kwd">readRDS</span><span class="hl std">(</span><span class="hl str">&quot;/DATA/usr/joris/git/SuRE_shared/Joris/SNP_project/180813_fdr_analysis/hepg2.sign.id.JvA180828.rds&quot;</span><span class="hl std">)</span>
<span class="hl kwd">length</span><span class="hl std">(DhitsH)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 14183
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#load Joris' list of expression-matched set of non-hits HepG2:</span>
<span class="hl std">DmatchH</span><span class="hl kwb">&lt;-</span><span class="hl kwd">readRDS</span><span class="hl std">(</span><span class="hl str">&quot;/DATA/usr/joris/git/SuRE_shared/Joris/SNP_project/180910_Matched_SNPs/hepg2.matched.control.180910.rds&quot;</span><span class="hl std">)</span>
<span class="hl kwd">length</span><span class="hl std">(DmatchH)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 14183
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#load and parse annotation file (also downloaded from GTEx v7 on 27 Jan 2018)</span>
<span class="hl com">#the parsing takes a few steps because this file has way too much information and is in .gtf format</span>
<span class="hl std">Ann</span><span class="hl kwb">&lt;-</span><span class="hl kwd">read.delim</span><span class="hl std">(</span><span class="hl str">&quot;./OtherData/GTEx_v7/gencode.v19.genes.v7.patched_contigs.gtf&quot;</span><span class="hl std">,</span> <span class="hl kwc">header</span><span class="hl std">=</span><span class="hl num">FALSE</span><span class="hl std">,</span> <span class="hl kwc">skip</span><span class="hl std">=</span><span class="hl num">6</span><span class="hl std">)</span>
<span class="hl std">Ann</span><span class="hl kwb">&lt;-</span><span class="hl std">Ann[Ann</span><span class="hl opt">$</span><span class="hl std">V3</span><span class="hl opt">==</span><span class="hl str">&quot;gene&quot;</span><span class="hl std">,]</span>
<span class="hl std">c1</span><span class="hl kwb">&lt;-</span><span class="hl kwd">regexpr</span><span class="hl std">(</span><span class="hl str">&quot;ENSG.*;&quot;</span><span class="hl std">, Ann</span><span class="hl opt">$</span><span class="hl std">V9)</span>
<span class="hl std">c2</span><span class="hl kwb">&lt;-</span><span class="hl kwd">regexpr</span><span class="hl std">(</span><span class="hl str">&quot;;&quot;</span><span class="hl std">, Ann</span><span class="hl opt">$</span><span class="hl std">V9)</span>
<span class="hl std">Ann</span><span class="hl opt">$</span><span class="hl std">gene_id</span><span class="hl kwb">&lt;-</span><span class="hl kwd">substr</span><span class="hl std">(Ann</span><span class="hl opt">$</span><span class="hl std">V9, c1, c2</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl std">)</span>
<span class="hl std">gene_name</span><span class="hl kwb">&lt;-</span><span class="hl kwd">regmatches</span><span class="hl std">(Ann</span><span class="hl opt">$</span><span class="hl std">V9,</span><span class="hl kwd">regexpr</span><span class="hl std">(</span><span class="hl str">&quot;gene_name.*;&quot;</span><span class="hl std">, Ann</span><span class="hl opt">$</span><span class="hl std">V9))</span>
<span class="hl std">c1</span><span class="hl kwb">&lt;-</span><span class="hl num">11</span>
<span class="hl std">c2</span><span class="hl kwb">&lt;-</span><span class="hl kwd">regexpr</span><span class="hl std">(</span><span class="hl str">&quot;;&quot;</span><span class="hl std">, gene_name)</span>
<span class="hl std">Ann</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl kwb">&lt;-</span><span class="hl kwd">substr</span><span class="hl std">(gene_name, c1, c2</span><span class="hl opt">-</span><span class="hl num">1</span><span class="hl std">)</span>
<span class="hl kwd">rm</span><span class="hl std">(gene_name)</span> <span class="hl com">#clean up</span>
<span class="hl std">Ann</span><span class="hl kwb">&lt;-</span><span class="hl std">Ann[,</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">5</span><span class="hl std">,</span><span class="hl num">7</span><span class="hl std">,</span><span class="hl num">10</span><span class="hl std">,</span><span class="hl num">11</span><span class="hl std">)]</span>
<span class="hl kwd">names</span><span class="hl std">(Ann)</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">&quot;chrom&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;start&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;end&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;orientation&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;gene_id&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;gene_name&quot;</span><span class="hl std">)</span>
<span class="hl kwd">dim</span><span class="hl std">(Ann)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 56202     6
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl kwd">head</span><span class="hl std">(Ann)</span>
</pre></div>

<div class="output"><pre class="knitr r">##    chrom start   end orientation           gene_id  gene_name
## 1      1 11869 14362           + ENSG00000223972.4    DDX11L1
## 7      1 14413 29553           - ENSG00000227232.4     WASH7P
## 22     1 29807 31109           + ENSG00000243485.2 MIR1302-11
## 27     1 34554 36081           - ENSG00000237613.2    FAM138A
## 32     1 52473 54936           + ENSG00000268020.2     OR4G4P
## 36     1 62948 63887           + ENSG00000240361.1    OR4G11P
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#load Gtex eQTL list for whole blood:</span>
<span class="hl std">Qb</span><span class="hl kwb">&lt;-</span><span class="hl kwd">read.delim</span><span class="hl std">(</span><span class="hl str">&quot;./OtherData/GTEx_v7/GTEx_Analysis_v7_eQTL/Whole_Blood.v7.signif_variant_gene_pairs.txt.gz&quot;</span><span class="hl std">,</span> <span class="hl kwc">head</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">,</span> <span class="hl kwc">as.is</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
<span class="hl com">#and add gene names:</span>
<span class="hl std">Qb</span><span class="hl kwb">&lt;-</span><span class="hl kwd">merge</span><span class="hl std">(Qb, Ann,</span> <span class="hl kwc">by</span><span class="hl std">=</span><span class="hl str">&quot;gene_id&quot;</span><span class="hl std">,</span> <span class="hl kwc">all.x</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">,</span> <span class="hl kwc">all.y</span><span class="hl std">=</span><span class="hl num">FALSE</span><span class="hl std">)</span>
<span class="hl std">Qb</span><span class="hl opt">$</span><span class="hl std">MLOG10P</span><span class="hl kwb">&lt;-</span> <span class="hl opt">-</span><span class="hl kwd">log10</span><span class="hl std">(Qb</span><span class="hl opt">$</span><span class="hl std">pval_nominal)</span> <span class="hl com">#useful for plotting of p.values, etc</span>
<span class="hl com">#reconstruct chrpos in Qb:</span>
<span class="hl std">Qb</span><span class="hl opt">$</span><span class="hl std">chrpos</span><span class="hl kwb">&lt;-</span><span class="hl kwd">gsub</span><span class="hl std">(</span><span class="hl str">&quot;^([^_]*_[^_]*)_.*&quot;</span><span class="hl std">,</span><span class="hl str">'\\1'</span><span class="hl std">,Qb</span><span class="hl opt">$</span><span class="hl std">variant_id)</span> <span class="hl com">#copied from Joris</span>
<span class="hl com">#also extract snp_pos for later:</span>
<span class="hl std">Qb</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl kwb">&lt;-</span><span class="hl kwd">as.integer</span><span class="hl std">(</span><span class="hl kwd">substr</span><span class="hl std">(Qb</span><span class="hl opt">$</span><span class="hl std">chrpos,</span> <span class="hl kwc">start</span><span class="hl std">=</span><span class="hl kwd">regexpr</span><span class="hl std">(</span><span class="hl str">&quot;_&quot;</span><span class="hl std">, Qb</span><span class="hl opt">$</span><span class="hl std">chrpos)</span><span class="hl opt">+</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl kwc">stop</span><span class="hl std">=</span><span class="hl num">100</span><span class="hl std">))</span>
<span class="hl com">##!!!!!!!!! THERE SEEMS TO BE A MISTAKE IN THE tss_distance column of Qb (as obtained from GTEx):</span>
<span class="hl com">#they correctly used the orientation of the gene to infer which end of the gene is the TSS;</span>
<span class="hl com">#but they forgot that the sign of the distance should also be flipped if orientation==&quot;-&quot;</span>
<span class="hl com">#as provided, if orientation==&quot;-&quot;, then tss_distance&lt;0 means that the SNP is *downstream* of the TSS</span>
<span class="hl com">#biologically this is not right, so let's calculate my own distance column:</span>
<span class="hl std">Qb</span><span class="hl opt">$</span><span class="hl std">snp2tss</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(Qb</span><span class="hl opt">$</span><span class="hl std">orientation</span><span class="hl opt">==</span><span class="hl str">&quot;-&quot;</span><span class="hl std">, Qb</span><span class="hl opt">$</span><span class="hl std">end</span><span class="hl opt">-</span><span class="hl std">Qb</span><span class="hl opt">$</span><span class="hl std">snp_pos, Qb</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl opt">-</span><span class="hl std">Qb</span><span class="hl opt">$</span><span class="hl std">start)</span>
<span class="hl kwd">dim</span><span class="hl std">(Qb)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 1052542      21
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl kwd">head</span><span class="hl std">(Qb)</span>
</pre></div>

<div class="output"><pre class="knitr r">##             gene_id           variant_id tss_distance ma_samples ma_count
## 1 ENSG00000000457.9  1_169658189_G_A_b37      -205219        248      323
## 2 ENSG00000000457.9 1_169659779_A_AT_b37      -203629        231      292
## 3 ENSG00000000457.9  1_169660002_G_T_b37      -203406        230      291
## 4 ENSG00000000457.9  1_169667355_T_A_b37      -196053        269      356
## 5 ENSG00000000457.9  1_169671088_A_C_b37      -192320         87       95
## 6 ENSG00000000457.9  1_169671117_G_A_b37      -192291         87       95
##        maf pval_nominal     slope  slope_se pval_nominal_threshold
## 1 0.437669  6.56587e-07 -0.132494 0.0260724            4.96987e-05
## 2 0.397820  1.37149e-05 -0.118443 0.0267904            4.96987e-05
## 3 0.395380  1.56351e-05 -0.116752 0.0265898            4.96987e-05
## 4 0.482385  1.63031e-07 -0.137874 0.0257083            4.96987e-05
## 5 0.128726  3.04860e-05  0.158556 0.0374471            4.96987e-05
## 6 0.128726  3.04860e-05  0.158556 0.0374471            4.96987e-05
##   min_pval_nominal   pval_beta chrom     start       end orientation
## 1      9.64392e-16 8.72938e-12     1 169818772 169863408           -
## 2      9.64392e-16 8.72938e-12     1 169818772 169863408           -
## 3      9.64392e-16 8.72938e-12     1 169818772 169863408           -
## 4      9.64392e-16 8.72938e-12     1 169818772 169863408           -
## 5      9.64392e-16 8.72938e-12     1 169818772 169863408           -
## 6      9.64392e-16 8.72938e-12     1 169818772 169863408           -
##   gene_name  MLOG10P      chrpos   snp_pos snp2tss
## 1     SCYL3 6.182708 1_169658189 169658189  205219
## 2     SCYL3 4.862807 1_169659779 169659779  203629
## 3     SCYL3 4.805899 1_169660002 169660002  203406
## 4     SCYL3 6.787730 1_169667355 169667355  196053
## 5     SCYL3 4.515900 1_169671088 169671088  192320
## 6     SCYL3 4.515900 1_169671117 169671117  192291
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#how many eQTL SNPs per eGene:</span>
<span class="hl kwd">mean</span><span class="hl std">(</span><span class="hl kwd">table</span><span class="hl std">(Qb</span><span class="hl opt">$</span><span class="hl std">gene_name))</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 121.5266
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#same for liver:</span>
<span class="hl std">Ql</span><span class="hl kwb">&lt;-</span><span class="hl kwd">read.delim</span><span class="hl std">(</span><span class="hl str">&quot;./OtherData/GTEx_v7/GTEx_Analysis_v7_eQTL/Liver.v7.signif_variant_gene_pairs.txt.gz&quot;</span><span class="hl std">,</span> <span class="hl kwc">head</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">,</span> <span class="hl kwc">as.is</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
<span class="hl com">#and add gene names:</span>
<span class="hl std">Ql</span><span class="hl kwb">&lt;-</span><span class="hl kwd">merge</span><span class="hl std">(Ql, Ann,</span> <span class="hl kwc">by</span><span class="hl std">=</span><span class="hl str">&quot;gene_id&quot;</span><span class="hl std">,</span> <span class="hl kwc">all.x</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">,</span> <span class="hl kwc">all.y</span><span class="hl std">=</span><span class="hl num">FALSE</span><span class="hl std">)</span>
<span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">MLOG10P</span><span class="hl kwb">&lt;-</span> <span class="hl opt">-</span><span class="hl kwd">log10</span><span class="hl std">(Ql</span><span class="hl opt">$</span><span class="hl std">pval_nominal)</span> <span class="hl com">#useful for plotting of p.values, etc</span>
<span class="hl com">#reconstruct chrpos in Ql:</span>
<span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">chrpos</span><span class="hl kwb">&lt;-</span><span class="hl kwd">gsub</span><span class="hl std">(</span><span class="hl str">&quot;^([^_]*_[^_]*)_.*&quot;</span><span class="hl std">,</span><span class="hl str">'\\1'</span><span class="hl std">,Ql</span><span class="hl opt">$</span><span class="hl std">variant_id)</span> <span class="hl com">#copied from Joris</span>
<span class="hl com">#also extract snp_pos for later:</span>
<span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl kwb">&lt;-</span><span class="hl kwd">as.integer</span><span class="hl std">(</span><span class="hl kwd">substr</span><span class="hl std">(Ql</span><span class="hl opt">$</span><span class="hl std">chrpos,</span> <span class="hl kwc">start</span><span class="hl std">=</span><span class="hl kwd">regexpr</span><span class="hl std">(</span><span class="hl str">&quot;_&quot;</span><span class="hl std">, Ql</span><span class="hl opt">$</span><span class="hl std">chrpos)</span><span class="hl opt">+</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl kwc">stop</span><span class="hl std">=</span><span class="hl num">100</span><span class="hl std">))</span>
<span class="hl com">##!!!!!!!!! THERE SEEMS TO BE A MISTAKE IN THE tss_distance column of Ql (as obtained from GTEx):</span>
<span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">snp2tss</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(Ql</span><span class="hl opt">$</span><span class="hl std">orientation</span><span class="hl opt">==</span><span class="hl str">&quot;-&quot;</span><span class="hl std">, Ql</span><span class="hl opt">$</span><span class="hl std">end</span><span class="hl opt">-</span><span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">snp_pos, Ql</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl opt">-</span><span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">start)</span>
<span class="hl kwd">dim</span><span class="hl std">(Ql)</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 323428     21
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl kwd">head</span><span class="hl std">(Ql)</span>
</pre></div>

<div class="output"><pre class="knitr r">##              gene_id          variant_id tss_distance ma_samples ma_count
## 1 ENSG00000000460.12 1_169579753_G_A_b37       -51492         15       15
## 2 ENSG00000000460.12 1_169581130_G_A_b37       -50115         29       31
## 3 ENSG00000000460.12 1_169584545_G_A_b37       -46700         15       15
## 4 ENSG00000000460.12 1_169587509_G_A_b37       -43736         15       15
## 5 ENSG00000000460.12 1_169590635_G_A_b37       -40610         14       14
## 6 ENSG00000000460.12 1_169591703_A_G_b37       -39542         29       31
##         maf pval_nominal    slope slope_se pval_nominal_threshold
## 1 0.0490196  1.26976e-06 0.925892 0.181119            1.77216e-05
## 2 0.1013070  1.10838e-07 0.682775 0.120617            1.77216e-05
## 3 0.0490196  1.26976e-06 0.925892 0.181119            1.77216e-05
## 4 0.0490196  1.26976e-06 0.925892 0.181119            1.77216e-05
## 5 0.0463576  1.42566e-06 0.944836 0.185801            1.77216e-05
## 6 0.1013070  1.10838e-07 0.682775 0.120617            1.77216e-05
##   min_pval_nominal   pval_beta chrom     start       end orientation
## 1      4.93938e-13 3.05848e-09     1 169631245 169823221           +
## 2      4.93938e-13 3.05848e-09     1 169631245 169823221           +
## 3      4.93938e-13 3.05848e-09     1 169631245 169823221           +
## 4      4.93938e-13 3.05848e-09     1 169631245 169823221           +
## 5      4.93938e-13 3.05848e-09     1 169631245 169823221           +
## 6      4.93938e-13 3.05848e-09     1 169631245 169823221           +
##   gene_name  MLOG10P      chrpos   snp_pos snp2tss
## 1  C1orf112 5.896278 1_169579753 169579753  -51492
## 2  C1orf112 6.955311 1_169581130 169581130  -50115
## 3  C1orf112 5.896278 1_169584545 169584545  -46700
## 4  C1orf112 5.896278 1_169587509 169587509  -43736
## 5  C1orf112 5.845984 1_169590635 169590635  -40610
## 6  C1orf112 6.955311 1_169591703 169591703  -39542
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#how many eQTL SNPs per eGene:</span>
<span class="hl kwd">mean</span><span class="hl std">(</span><span class="hl kwd">table</span><span class="hl std">(Ql</span><span class="hl opt">$</span><span class="hl std">gene_name))</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] 80.857
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#merge data:</span>
<span class="hl std">DQb</span><span class="hl kwb">&lt;-</span><span class="hl kwd">merge</span><span class="hl std">(D[,</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">10</span><span class="hl std">,</span><span class="hl num">12</span><span class="hl std">,</span><span class="hl num">14</span><span class="hl std">,</span><span class="hl num">26</span><span class="hl std">)], Qb[,</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">7</span><span class="hl std">,</span><span class="hl num">8</span><span class="hl std">,</span><span class="hl num">13</span><span class="hl opt">:</span><span class="hl num">21</span><span class="hl std">)],</span> <span class="hl kwc">by</span><span class="hl std">=</span><span class="hl str">&quot;chrpos&quot;</span><span class="hl std">)</span>
<span class="hl std">DQl</span><span class="hl kwb">&lt;-</span><span class="hl kwd">merge</span><span class="hl std">(D[,</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">16</span><span class="hl std">,</span><span class="hl num">18</span><span class="hl std">,</span><span class="hl num">20</span><span class="hl std">,</span><span class="hl num">26</span><span class="hl std">)], Ql[,</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">7</span><span class="hl std">,</span><span class="hl num">8</span><span class="hl std">,</span><span class="hl num">13</span><span class="hl opt">:</span><span class="hl num">21</span><span class="hl std">)],</span> <span class="hl kwc">by</span><span class="hl std">=</span><span class="hl str">&quot;chrpos&quot;</span><span class="hl std">)</span>

<span class="hl com">#calculate concordance between SuRE and eQTL signs:</span>
<span class="hl std">DQb</span><span class="hl opt">$</span><span class="hl std">concordant</span> <span class="hl kwb">&lt;-</span> <span class="hl std">DQb</span><span class="hl opt">$</span><span class="hl std">slope</span> <span class="hl opt">*</span> <span class="hl kwd">log2</span><span class="hl std">(DQb</span><span class="hl opt">$</span><span class="hl std">k562.alt.mean</span><span class="hl opt">/</span><span class="hl std">DQb</span><span class="hl opt">$</span><span class="hl std">k562.ref.mean)</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span>
<span class="hl std">DQl</span><span class="hl opt">$</span><span class="hl std">concordant</span> <span class="hl kwb">&lt;-</span> <span class="hl std">DQl</span><span class="hl opt">$</span><span class="hl std">slope</span> <span class="hl opt">*</span> <span class="hl kwd">log2</span><span class="hl std">(DQl</span><span class="hl opt">$</span><span class="hl std">hepg2.alt.mean</span><span class="hl opt">/</span><span class="hl std">DQl</span><span class="hl opt">$</span><span class="hl std">hepg2.ref.mean)</span> <span class="hl opt">&gt;</span> <span class="hl num">0</span>
</pre></div>

<p></div></div></p>

<p><b>Conclusion:</b>Data loaded: ~31M SNPs in GTEx, 6.6M in SuRE set, 1M in Whole Blood eQTL set, 0.3M in Liver eQTL set. </p><br>





<h3>Plot SuRE and eQTL scores around some interesting eGenes, K562</h3>
<p>Possible figure panels for the SNP-SuRE paper. With nice gene annotations generated by Gviz. For K562 we have the option to show a DNaseI track, so it is plotted here as well. </p>

<div class="chunk" id="html-Plot_SuRE_around_eGenes_WholeBlood_K562"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl std">PlotGenes</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">&quot;NFX1&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;LYZ&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;YEATS4&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;COL9A3&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;NEIL2&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;XPNPEP2&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;ABCC11&quot;</span><span class="hl std">)</span>

<span class="hl std">pplotcut</span><span class="hl kwb">&lt;-</span><span class="hl num">0.1</span> <span class="hl com">#pvalue cutoff of SuRE data that is taken into account when choosing the y-axis limit</span>
<span class="hl std">CrdList</span><span class="hl kwb">&lt;-</span><span class="hl kwd">list</span><span class="hl std">()</span> <span class="hl com">#to memorize plotting ranges -&gt; for consistency in case of COL9A3 in liver in next code chunk</span>

<span class="hl kwa">for</span><span class="hl std">(g</span> <span class="hl kwa">in</span> <span class="hl std">PlotGenes)</span>
  <span class="hl std">{</span><span class="hl kwd">print</span><span class="hl std">(g)</span>
   <span class="hl std">Qbs</span><span class="hl kwb">&lt;-</span><span class="hl std">Qb[Qb</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g,]</span> <span class="hl com">#data subsets for this eGene</span>
   <span class="hl std">DQbs</span><span class="hl kwb">&lt;-</span><span class="hl std">DQb[DQb</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g,]</span>
   <span class="hl com">#get IDs of all SNPs that are SuRE hit:</span>
   <span class="hl std">w</span><span class="hl kwb">&lt;-</span><span class="hl kwd">which</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">concordant</span> <span class="hl opt">&amp;</span> <span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">SNP_ID</span> <span class="hl opt">%in%</span> <span class="hl std">DhitsK))</span>
   <span class="hl kwd">print</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">SNP_ID[w])</span> <span class="hl com">#print these SNPs</span>

   <span class="hl com">##first plot SuRE data using regular plot() functions, in a separate PDF for each eGene:</span>
   <span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_SuRE_eQTL_overlap_K562_&quot;</span><span class="hl std">, g,</span> <span class="hl str">&quot;_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
       <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">6</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">3.5</span><span class="hl std">,</span> <span class="hl kwc">onefile</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
   <span class="hl com">#layout parameters:</span>
   <span class="hl std">m</span><span class="hl kwb">&lt;-</span><span class="hl kwd">matrix</span><span class="hl std">(</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">3</span><span class="hl std">,</span><span class="hl num">3</span><span class="hl std">))</span> <span class="hl com">#layout matrix</span>
   <span class="hl kwd">layout</span><span class="hl std">(m)</span>
   <span class="hl kwd">par</span><span class="hl std">(</span><span class="hl kwc">mar</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">mgp</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1.8</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">oma</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">),</span> <span class="hl kwc">lend</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">)</span>

   <span class="hl com">#determine genomic coordinate range:</span>
   <span class="hl std">gstart</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">orientation[</span><span class="hl num">1</span><span class="hl std">]</span><span class="hl opt">==</span><span class="hl str">&quot;+&quot;</span><span class="hl std">, DQbs</span><span class="hl opt">$</span><span class="hl std">start[</span><span class="hl num">1</span><span class="hl std">], DQbs</span><span class="hl opt">$</span><span class="hl std">end[</span><span class="hl num">1</span><span class="hl std">])</span>
   <span class="hl std">gend</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">orientation[</span><span class="hl num">1</span><span class="hl std">]</span><span class="hl opt">==</span><span class="hl str">&quot;-&quot;</span><span class="hl std">, DQbs</span><span class="hl opt">$</span><span class="hl std">start[</span><span class="hl num">1</span><span class="hl std">], DQbs</span><span class="hl opt">$</span><span class="hl std">end[</span><span class="hl num">1</span><span class="hl std">])</span>
   <span class="hl std">win</span><span class="hl kwb">&lt;-</span><span class="hl kwd">range</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">snp_pos, gstart, gend)</span>
   <span class="hl std">CrdList[[g]]</span><span class="hl kwb">&lt;-</span><span class="hl std">win</span> <span class="hl com">#keep coordinates</span>
   <span class="hl std">chr</span><span class="hl kwb">&lt;-</span><span class="hl kwd">unique</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">chrom)</span>
   <span class="hl std">chr</span><span class="hl kwb">&lt;-</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;chr&quot;</span><span class="hl std">, chr)</span>
   <span class="hl kwa">if</span><span class="hl std">(</span><span class="hl kwd">length</span><span class="hl std">(chr)</span><span class="hl opt">&gt;</span><span class="hl num">1</span><span class="hl std">){</span><span class="hl kwd">stop</span><span class="hl std">(</span><span class="hl str">&quot;multiple chromosomes&quot;</span><span class="hl std">)}</span>
   <span class="hl com">#SuRE SNP scores:</span>
   <span class="hl std">rr</span><span class="hl kwb">&lt;-</span><span class="hl std">DQbs</span><span class="hl opt">$</span><span class="hl std">k562.ref.mean</span>
   <span class="hl std">aa</span><span class="hl kwb">&lt;-</span><span class="hl std">DQbs</span><span class="hl opt">$</span><span class="hl std">k562.alt.mean</span>
   <span class="hl std">sylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">15</span><span class="hl std">, aa[DQbs</span><span class="hl opt">$</span><span class="hl std">k562.wilcox.p.value</span><span class="hl opt">&lt;</span><span class="hl std">pplotcut],rr[DQbs</span><span class="hl opt">$</span><span class="hl std">k562.wilcox.p.value</span><span class="hl opt">&lt;</span><span class="hl std">pplotcut]),</span> <span class="hl kwc">na.rm</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">))</span> <span class="hl com">#SuRE y-axis range</span>

   <span class="hl com">#SuRE plot: colors show concordance/discordance (see color choice above)</span>
   <span class="hl std">segcol</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">concordant, COLc, COLd)</span>
   <span class="hl kwd">plot</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">snp_pos, rr,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=sylim,</span> <span class="hl kwc">xaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">segments</span><span class="hl std">(</span><span class="hl kwc">x0</span><span class="hl std">=DQbs</span><span class="hl opt">$</span><span class="hl std">snp_pos,</span> <span class="hl kwc">y0</span><span class="hl std">=rr,</span> <span class="hl kwc">y1</span><span class="hl std">=aa,</span> <span class="hl kwc">col</span><span class="hl std">=segcol,</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl opt">-</span><span class="hl kwd">log10</span><span class="hl std">(DQbs</span><span class="hl opt">$</span><span class="hl std">k562.wilcox.p.value)</span><span class="hl opt">*</span><span class="hl std">p2width)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl str">&quot;SuRE signal&quot;</span><span class="hl std">,</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">2.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.8</span><span class="hl std">)</span>
   <span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topleft&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">paste</span><span class="hl std">(g,</span> <span class="hl str">&quot;(K562)&quot;</span><span class="hl std">),</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">1.2</span><span class="hl std">)</span>
   <span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topright&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;coverage = &quot;</span><span class="hl std">,</span> <span class="hl kwd">nrow</span><span class="hl std">(DQbs),</span> <span class="hl str">&quot;/&quot;</span><span class="hl std">,</span> <span class="hl kwd">table</span><span class="hl std">(Qb</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g)[</span><span class="hl str">&quot;TRUE&quot;</span><span class="hl std">]),</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>

   <span class="hl com">#eQTL scores:</span>
   <span class="hl std">eylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(Qbs</span><span class="hl opt">$</span><span class="hl std">MLOG10P))</span>
   <span class="hl kwd">plot</span><span class="hl std">(Qbs</span><span class="hl opt">$</span><span class="hl std">snp_pos, Qbs</span><span class="hl opt">$</span><span class="hl std">MLOG10P,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=eylim,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;h&quot;</span><span class="hl std">,</span> <span class="hl kwc">ylab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">xaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl str">&quot;eQTL&quot;</span><span class="hl std">,</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">2.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.8</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl kwd">expression</span><span class="hl std">(</span><span class="hl str">&quot;log&quot;</span><span class="hl std">[</span><span class="hl num">10</span><span class="hl std">]</span><span class="hl opt">*</span><span class="hl str">&quot;(P-value)&quot;</span><span class="hl std">),</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">1.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.7</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl kwd">paste</span><span class="hl std">(</span><span class="hl str">&quot;position on&quot;</span><span class="hl std">,chr),</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl kwc">outer</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">1.8</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.7</span><span class="hl std">)</span>

   <span class="hl com">#room for Gviz annotation track:</span>
   <span class="hl std">aylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(Qbs</span><span class="hl opt">$</span><span class="hl std">MLOG10P))</span>
   <span class="hl kwd">plot</span><span class="hl std">(Qbs</span><span class="hl opt">$</span><span class="hl std">snp_pos, Qbs</span><span class="hl opt">$</span><span class="hl std">MLOG10P,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=aylim,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">ylab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">yaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">arrows</span><span class="hl std">(</span><span class="hl kwc">x0</span><span class="hl std">=gstart,</span> <span class="hl kwc">y0</span><span class="hl std">=aylim[</span><span class="hl num">2</span><span class="hl std">]</span><span class="hl opt">/</span><span class="hl num">10</span><span class="hl std">,</span> <span class="hl kwc">x1</span><span class="hl std">=gend,</span> <span class="hl kwc">length</span><span class="hl std">=</span><span class="hl num">0.05</span><span class="hl std">,</span> <span class="hl kwc">angle</span><span class="hl std">=</span><span class="hl num">30</span><span class="hl std">,</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl num">3</span><span class="hl std">,</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl str">&quot;darkgrey&quot;</span><span class="hl std">)</span> <span class="hl com">#plot eGene for later checks</span>
   <span class="hl std">xlim.full</span><span class="hl kwb">&lt;-</span><span class="hl kwd">round</span><span class="hl std">(</span><span class="hl kwd">par</span><span class="hl std">(</span><span class="hl str">&quot;usr&quot;</span><span class="hl std">)[</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">2</span><span class="hl std">)])</span> <span class="hl com">#actual full plotting window, to be passed on to gviz functions.</span>

   <span class="hl kwd">dev.off</span><span class="hl std">()</span> <span class="hl com">#close SuRE/eQTL/frame plot </span>

   <span class="hl com">##Now generate gene track + DNase data track using gviz:</span>
   <span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_Gviz_annotation_K562_&quot;</span><span class="hl std">, g,</span> <span class="hl str">&quot;_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
       <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">12.2</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">1.1</span><span class="hl std">)</span> <span class="hl com">#broad for full locus</span>

   <span class="hl std">biomTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">BiomartGeneRegionTrack</span><span class="hl std">(</span><span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
                                    <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
                                    <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
                                    <span class="hl kwc">collapseTranscripts</span><span class="hl std">=</span><span class="hl str">&quot;meta&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">transcriptAnnotation</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span>
                                    <span class="hl kwc">cex.group</span><span class="hl std">=</span><span class="hl num">0.5</span><span class="hl std">,</span>
                                    <span class="hl kwc">shape</span><span class="hl std">=</span><span class="hl str">&quot;smallArrow&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">col.line</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span> <span class="hl kwc">fill</span><span class="hl std">=</span><span class="hl str">&quot;darkred&quot;</span><span class="hl std">,</span>
                                    <span class="hl com">#background.panel = &quot;#EEEEEE&quot;,</span>
                                    <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;Genes&quot;</span><span class="hl std">)</span>
  <span class="hl std">GR</span><span class="hl kwb">&lt;-</span><span class="hl kwd">makeGRangesFromDataFrame</span><span class="hl std">(</span><span class="hl kwd">data.frame</span><span class="hl std">(</span><span class="hl kwc">chr</span><span class="hl std">=chr,</span> <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span> <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span> <span class="hl kwc">score</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">))</span>
  <span class="hl std">DNase</span><span class="hl kwb">&lt;-</span><span class="hl kwd">import</span><span class="hl std">(URL.DNase.K562,</span> <span class="hl kwc">which</span><span class="hl std">=GR)</span> <span class="hl com">#grab only relevant region </span>
  <span class="hl std">dnaseTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">DataTrack</span><span class="hl std">(</span><span class="hl kwc">range</span><span class="hl std">=DNase,</span>
                            <span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
                            <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
                            <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
                            <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;h&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">ylim</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">1000</span><span class="hl std">),</span>
                            <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl str">&quot;darkgreen&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;DNaseI&quot;</span><span class="hl std">)</span>
  <span class="hl com">#optional TF ChIP-seq track:</span>
  <span class="hl com">#TFs&lt;-import.bed(URL.TF.K562, which=GR, genome=&quot;hg19&quot;)</span>
  <span class="hl com">#tfTrack&lt;- AnnotationTrack(TFs, </span>
   <span class="hl com">#                         genome=&quot;hg19&quot;,</span>
  <span class="hl com">#                          chromosome=chr, </span>
  <span class="hl com">#                          start=xlim.full[1], </span>
  <span class="hl com">#                          end=xlim.full[2], </span>
  <span class="hl com">#                          shape=&quot;box&quot;, fill=&quot;black&quot;, lwd=0, min.height=0.1, min.width=0.1, name=&quot;TF&quot;</span>
  <span class="hl com">#                         )</span>


  <span class="hl com">#create simple full-width bar to facilatate alignment in bottom frame of SuRE/eQTL plot:</span>
  <span class="hl std">aTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">AnnotationTrack</span><span class="hl std">(</span><span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span> <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl kwd">diff</span><span class="hl std">(xlim.full),</span> <span class="hl kwc">strand</span><span class="hl std">=</span><span class="hl str">&quot;*&quot;</span><span class="hl std">,</span> <span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span> <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span> <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;aln&quot;</span><span class="hl std">)</span>

  <span class="hl kwd">plotTracks</span><span class="hl std">(</span><span class="hl kwd">list</span><span class="hl std">(biomTrack, dnaseTrack, aTrack),</span>
             <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
             <span class="hl kwc">from</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
             <span class="hl kwc">to</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
             <span class="hl kwc">sizes</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl num">0.3</span><span class="hl std">),</span>
             <span class="hl kwc">fontcolor.title</span><span class="hl std">=</span><span class="hl str">&quot;black&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">col.border.title</span><span class="hl std">=</span><span class="hl str">&quot;transparent&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">background.title</span><span class="hl std">=</span><span class="hl str">&quot;transparent&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">col.axis</span><span class="hl std">=</span><span class="hl str">&quot;black&quot;</span>
             <span class="hl std">)</span>


  <span class="hl kwd">dev.off</span><span class="hl std">()</span> <span class="hl com">#close Gviz PDF</span>


<span class="hl std">}</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;NFX1&quot;
## [1] &quot;rs4879682&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;LYZ&quot;
## [1] &quot;rs623853&quot; &quot;rs554591&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;YEATS4&quot;
## [1] &quot;rs623853&quot; &quot;rs554591&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;COL9A3&quot;
## [1] &quot;rs1018439&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;NEIL2&quot;
## [1] &quot;rs7838131&quot;  &quot;rs11996277&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;XPNPEP2&quot;
## character(0)
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;ABCC11&quot;
## character(0)
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#add legends in empty plot, size matched:</span>
<span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_SuRE_eQTL_overlap_K562_LEGEND_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
    <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">6</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">3.5</span><span class="hl std">,</span> <span class="hl kwc">onefile</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
<span class="hl kwd">layout</span><span class="hl std">(m)</span>
<span class="hl kwd">par</span><span class="hl std">(</span><span class="hl kwc">mar</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">mgp</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1.8</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">oma</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">),</span> <span class="hl kwc">lend</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">)</span>
<span class="hl kwd">plot.new</span><span class="hl std">()</span> <span class="hl com">#open plot</span>
<span class="hl std">Leg</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">6</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">8</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">10</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">)</span>
<span class="hl std">Lwd</span><span class="hl kwb">&lt;-</span><span class="hl std">Leg</span><span class="hl opt">*</span><span class="hl std">p2width</span>
<span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topleft&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=Leg,</span> <span class="hl kwc">lwd</span><span class="hl std">=Lwd,</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">title</span><span class="hl std">=</span><span class="hl str">&quot;SuRE -log10(wilcox.p)&quot;</span><span class="hl std">,</span> <span class="hl kwc">xpd</span><span class="hl std">=</span><span class="hl num">NA</span><span class="hl std">)</span>
<span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topright&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">NA</span><span class="hl std">,</span> <span class="hl str">&quot;concordant&quot;</span><span class="hl std">,</span> <span class="hl num">NA</span><span class="hl std">,</span> <span class="hl str">&quot;discordant&quot;</span><span class="hl std">),</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">NA</span><span class="hl std">, COLc,</span> <span class="hl num">NA</span><span class="hl std">, COLd),</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl num">20</span><span class="hl std">,</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">title</span><span class="hl std">=</span><span class="hl str">&quot;SuRE vs eQTL&quot;</span><span class="hl std">)</span>
<span class="hl kwd">plot.new</span><span class="hl std">()</span> <span class="hl com">#fill up </span>
<span class="hl kwd">plot.new</span><span class="hl std">()</span> <span class="hl com">#fill up </span>
<span class="hl kwd">dev.off</span><span class="hl std">()</span>
</pre></div>

<div class="output"><pre class="knitr r">## png 
##   2
</pre></div>

<p></div></div></p>

<p><b>Note:</b>Gviz plots can now be pasted into bottom frame of SuRE/eQTL plots using Illustrator. The width of the former should be adjusted until the light-blue bar fits <i>exactly</i> in the frame. The horizontal grey arrow marks the position of the eGene and may be removed. The alignment bar, and optionally the DNaseI track, may then also be removed manually.</p>

<h3>Plot SuRE and eQTL scores and annotations around some interesting eGenes, HepG2</h3>

<p>Examples to be highlighted in the paper.</p>

<div class="chunk" id="html-Plot_SuRE_around_eGenes_WholeBlood_HepG2"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl std">PlotGenes</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">&quot;LYZ&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;YEATS4&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;COL9A3&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;XPNPEP2&quot;</span><span class="hl std">,</span> <span class="hl str">&quot;ABCC11&quot;</span><span class="hl std">)</span>

<span class="hl kwa">for</span><span class="hl std">(g</span> <span class="hl kwa">in</span> <span class="hl std">PlotGenes)</span>
  <span class="hl std">{</span><span class="hl kwd">print</span><span class="hl std">(g)</span>
   <span class="hl std">Qls</span><span class="hl kwb">&lt;-</span><span class="hl std">Ql[Ql</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g,]</span> <span class="hl com">#data subsets for this eGene</span>
   <span class="hl std">DQls</span><span class="hl kwb">&lt;-</span><span class="hl std">DQl[DQl</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g,]</span>
   <span class="hl com">#get IDs of all SNPs that are SuRE hit:</span>
   <span class="hl std">w</span><span class="hl kwb">&lt;-</span><span class="hl kwd">which</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">concordant</span> <span class="hl opt">&amp;</span> <span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">SNP_ID</span> <span class="hl opt">%in%</span> <span class="hl std">DhitsH))</span>
   <span class="hl kwd">print</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">SNP_ID[w])</span> <span class="hl com">#print these SNPs</span>

   <span class="hl com">##first plot SuRE data using regular plot() functions, in a separate PDF for each eGene:</span>
   <span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_SuRE_eQTL_overlap_HepG2_&quot;</span><span class="hl std">, g,</span> <span class="hl str">&quot;_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
       <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">6</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">3.5</span><span class="hl std">,</span> <span class="hl kwc">onefile</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
   <span class="hl com">#layout parameters:</span>
   <span class="hl std">m</span><span class="hl kwb">&lt;-</span><span class="hl kwd">matrix</span><span class="hl std">(</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">3</span><span class="hl std">,</span><span class="hl num">3</span><span class="hl std">))</span> <span class="hl com">#layout matrix</span>
   <span class="hl kwd">layout</span><span class="hl std">(m)</span>
   <span class="hl kwd">par</span><span class="hl std">(</span><span class="hl kwc">mar</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">mgp</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1.8</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">oma</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">),</span> <span class="hl kwc">lend</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">)</span>

   <span class="hl com">#determine genomic coordinate range:</span>
   <span class="hl std">gstart</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">orientation[</span><span class="hl num">1</span><span class="hl std">]</span><span class="hl opt">==</span><span class="hl str">&quot;+&quot;</span><span class="hl std">, DQls</span><span class="hl opt">$</span><span class="hl std">start[</span><span class="hl num">1</span><span class="hl std">], DQls</span><span class="hl opt">$</span><span class="hl std">end[</span><span class="hl num">1</span><span class="hl std">])</span>
   <span class="hl std">gend</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">orientation[</span><span class="hl num">1</span><span class="hl std">]</span><span class="hl opt">==</span><span class="hl str">&quot;-&quot;</span><span class="hl std">, DQls</span><span class="hl opt">$</span><span class="hl std">start[</span><span class="hl num">1</span><span class="hl std">], DQls</span><span class="hl opt">$</span><span class="hl std">end[</span><span class="hl num">1</span><span class="hl std">])</span>
   <span class="hl std">win</span><span class="hl kwb">&lt;-</span><span class="hl kwd">range</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">snp_pos, gstart, gend)</span>
   <span class="hl kwa">if</span><span class="hl std">(g</span><span class="hl opt">==</span><span class="hl str">&quot;COL9A3&quot;</span><span class="hl std">){win</span><span class="hl kwb">&lt;-</span><span class="hl kwd">unlist</span><span class="hl std">(CrdList[g])}</span> <span class="hl com">#for COL9A3 use same range as for K562</span>
   <span class="hl std">chr</span><span class="hl kwb">&lt;-</span><span class="hl kwd">unique</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">chrom)</span>
   <span class="hl std">chr</span><span class="hl kwb">&lt;-</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;chr&quot;</span><span class="hl std">, chr)</span>
   <span class="hl kwa">if</span><span class="hl std">(</span><span class="hl kwd">length</span><span class="hl std">(chr)</span><span class="hl opt">&gt;</span><span class="hl num">1</span><span class="hl std">){</span><span class="hl kwd">stop</span><span class="hl std">(</span><span class="hl str">&quot;multiple chromosomes&quot;</span><span class="hl std">)}</span>
   <span class="hl com">#SuRE SNP scores:</span>
   <span class="hl std">rr</span><span class="hl kwb">&lt;-</span><span class="hl std">DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.ref.mean</span>
   <span class="hl std">aa</span><span class="hl kwb">&lt;-</span><span class="hl std">DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.alt.mean</span>
   <span class="hl std">sylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">15</span><span class="hl std">, aa[DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.wilcox.p.value</span><span class="hl opt">&lt;</span><span class="hl std">pplotcut],rr[DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.wilcox.p.value</span><span class="hl opt">&lt;</span><span class="hl std">pplotcut]),</span> <span class="hl kwc">na.rm</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">))</span> <span class="hl com">#SuRE y-axis range</span>

   <span class="hl com">#SuRE plot: colors show concordance/discordance (see color choice above)</span>
   <span class="hl std">segcol</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">concordant, COLc, COLd)</span>
   <span class="hl kwd">plot</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">snp_pos, rr,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=sylim,</span> <span class="hl kwc">xaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">segments</span><span class="hl std">(</span><span class="hl kwc">x0</span><span class="hl std">=DQls</span><span class="hl opt">$</span><span class="hl std">snp_pos,</span> <span class="hl kwc">y0</span><span class="hl std">=rr,</span> <span class="hl kwc">y1</span><span class="hl std">=aa,</span> <span class="hl kwc">col</span><span class="hl std">=segcol,</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl opt">-</span><span class="hl kwd">log10</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.wilcox.p.value)</span><span class="hl opt">*</span><span class="hl std">p2width)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl str">&quot;SuRE signal&quot;</span><span class="hl std">,</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">2.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.8</span><span class="hl std">)</span>
   <span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topleft&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">paste</span><span class="hl std">(g,</span> <span class="hl str">&quot;(HepG2)&quot;</span><span class="hl std">),</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">1.2</span><span class="hl std">)</span>
   <span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topright&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;coverage = &quot;</span><span class="hl std">,</span> <span class="hl kwd">nrow</span><span class="hl std">(DQls),</span> <span class="hl str">&quot;/&quot;</span><span class="hl std">,</span> <span class="hl kwd">table</span><span class="hl std">(Ql</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g)[</span><span class="hl str">&quot;TRUE&quot;</span><span class="hl std">]),</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>

   <span class="hl com">#eQTL scores:</span>
   <span class="hl std">eylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P))</span>
   <span class="hl kwd">plot</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">snp_pos, Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=eylim,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;h&quot;</span><span class="hl std">,</span> <span class="hl kwc">ylab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">xaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl str">&quot;eQTL&quot;</span><span class="hl std">,</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">2.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.8</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl kwd">expression</span><span class="hl std">(</span><span class="hl str">&quot;log&quot;</span><span class="hl std">[</span><span class="hl num">10</span><span class="hl std">]</span><span class="hl opt">*</span><span class="hl str">&quot;(P-value)&quot;</span><span class="hl std">),</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">1.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.7</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl kwd">paste</span><span class="hl std">(</span><span class="hl str">&quot;position on&quot;</span><span class="hl std">,chr),</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl kwc">outer</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">1.8</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.7</span><span class="hl std">)</span>

   <span class="hl com">#room for Gviz annotation track:</span>
   <span class="hl std">aylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P))</span>
   <span class="hl kwd">plot</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">snp_pos, Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=aylim,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">ylab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">yaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">arrows</span><span class="hl std">(</span><span class="hl kwc">x0</span><span class="hl std">=gstart,</span> <span class="hl kwc">y0</span><span class="hl std">=aylim[</span><span class="hl num">2</span><span class="hl std">]</span><span class="hl opt">/</span><span class="hl num">10</span><span class="hl std">,</span> <span class="hl kwc">x1</span><span class="hl std">=gend,</span> <span class="hl kwc">length</span><span class="hl std">=</span><span class="hl num">0.05</span><span class="hl std">,</span> <span class="hl kwc">angle</span><span class="hl std">=</span><span class="hl num">30</span><span class="hl std">,</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl num">3</span><span class="hl std">,</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl str">&quot;darkgrey&quot;</span><span class="hl std">)</span> <span class="hl com">#plot eGene for later checks</span>
   <span class="hl std">xlim.full</span><span class="hl kwb">&lt;-</span><span class="hl kwd">round</span><span class="hl std">(</span><span class="hl kwd">par</span><span class="hl std">(</span><span class="hl str">&quot;usr&quot;</span><span class="hl std">)[</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">2</span><span class="hl std">)])</span> <span class="hl com">#actual full plotting window, to be passed on to gviz functions.</span>

   <span class="hl kwd">dev.off</span><span class="hl std">()</span> <span class="hl com">#close SuRE/eQTL/frame plot </span>

   <span class="hl com">##Now generate gene track using gviz:</span>
   <span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_Gviz_annotation_HepG2_&quot;</span><span class="hl std">, g,</span> <span class="hl str">&quot;_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
       <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">12.2</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">1.1</span><span class="hl std">)</span> <span class="hl com">#broad for full locus</span>

   <span class="hl std">biomTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">BiomartGeneRegionTrack</span><span class="hl std">(</span><span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
                                    <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
                                    <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
                                    <span class="hl kwc">collapseTranscripts</span><span class="hl std">=</span><span class="hl str">&quot;meta&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">transcriptAnnotation</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span>
                                    <span class="hl kwc">cex.group</span><span class="hl std">=</span><span class="hl num">0.5</span><span class="hl std">,</span>
                                    <span class="hl kwc">shape</span><span class="hl std">=</span><span class="hl str">&quot;smallArrow&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">col.line</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span> <span class="hl kwc">fill</span><span class="hl std">=</span><span class="hl str">&quot;darkred&quot;</span><span class="hl std">,</span>
                                    <span class="hl com">#background.panel = &quot;#EEEEEE&quot;,</span>
                                    <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;Genes&quot;</span><span class="hl std">)</span>
  <span class="hl std">GR</span><span class="hl kwb">&lt;-</span><span class="hl kwd">makeGRangesFromDataFrame</span><span class="hl std">(</span><span class="hl kwd">data.frame</span><span class="hl std">(</span><span class="hl kwc">chr</span><span class="hl std">=chr,</span> <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span> <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span> <span class="hl kwc">score</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">))</span>

  <span class="hl com">#DNase track:</span>
  <span class="hl std">DNase</span><span class="hl kwb">&lt;-</span><span class="hl kwd">import</span><span class="hl std">(URL.DNase.HepG2,</span> <span class="hl kwc">which</span><span class="hl std">=GR)</span> <span class="hl com">#grab only relevant region </span>
  <span class="hl std">dnaseTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">DataTrack</span><span class="hl std">(</span><span class="hl kwc">range</span><span class="hl std">=DNase,</span>
                            <span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
                            <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
                            <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
                            <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;h&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">ylim</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">200</span><span class="hl std">),</span>
                            <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl str">&quot;darkgreen&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;DNaseI&quot;</span><span class="hl std">)</span> <span class="hl com">#scale adjusted to lower dynamic range compared to K562</span>

  <span class="hl com">#create simple full-width bar to facilatate alignment in bottom frame of SuRE/eQTL plot:</span>
  <span class="hl std">aTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">AnnotationTrack</span><span class="hl std">(</span><span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span> <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl kwd">diff</span><span class="hl std">(xlim.full),</span> <span class="hl kwc">strand</span><span class="hl std">=</span><span class="hl str">&quot;*&quot;</span><span class="hl std">,</span> <span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span> <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span> <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;aln&quot;</span><span class="hl std">)</span>

  <span class="hl kwd">plotTracks</span><span class="hl std">(</span><span class="hl kwd">list</span><span class="hl std">(biomTrack, dnaseTrack, aTrack),</span>
             <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
             <span class="hl kwc">from</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
             <span class="hl kwc">to</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
             <span class="hl kwc">sizes</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl num">0.3</span><span class="hl std">),</span>
             <span class="hl kwc">fontcolor.title</span><span class="hl std">=</span><span class="hl str">&quot;black&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">col.border.title</span><span class="hl std">=</span><span class="hl str">&quot;transparent&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">background.title</span><span class="hl std">=</span><span class="hl str">&quot;transparent&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">col.axis</span><span class="hl std">=</span><span class="hl str">&quot;black&quot;</span>
             <span class="hl std">)</span>


  <span class="hl kwd">dev.off</span><span class="hl std">()</span> <span class="hl com">#close Gviz PDF</span>


<span class="hl std">}</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;LYZ&quot;
## character(0)
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;YEATS4&quot;
## [1] &quot;rs623853&quot; &quot;rs554591&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;COL9A3&quot;
## [1] &quot;rs1018439&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;XPNPEP2&quot;
## [1] &quot;rs3788853&quot;
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;ABCC11&quot;
## [1] &quot;rs11863133&quot; &quot;rs11866312&quot; &quot;rs11863233&quot; &quot;rs11863236&quot;
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">#add legends in empty plot, size matched:</span>
<span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_SuRE_eQTL_overlap_HepG2_LEGEND_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
    <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">6</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">3.5</span><span class="hl std">,</span> <span class="hl kwc">onefile</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
<span class="hl kwd">layout</span><span class="hl std">(m)</span>
<span class="hl kwd">par</span><span class="hl std">(</span><span class="hl kwc">mar</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">mgp</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1.8</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">oma</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">),</span> <span class="hl kwc">lend</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">)</span>
<span class="hl kwd">plot.new</span><span class="hl std">()</span> <span class="hl com">#open plot</span>
<span class="hl std">Leg</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">2</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">6</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">8</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">,</span><span class="hl num">10</span><span class="hl std">,</span><span class="hl num">NA</span><span class="hl std">)</span>
<span class="hl std">Lwd</span><span class="hl kwb">&lt;-</span><span class="hl std">Leg</span><span class="hl opt">*</span><span class="hl std">p2width</span>
<span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topleft&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=Leg,</span> <span class="hl kwc">lwd</span><span class="hl std">=Lwd,</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">title</span><span class="hl std">=</span><span class="hl str">&quot;SuRE -log10(wilcox.p)&quot;</span><span class="hl std">,</span> <span class="hl kwc">xpd</span><span class="hl std">=</span><span class="hl num">NA</span><span class="hl std">)</span>
<span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topright&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">NA</span><span class="hl std">,</span> <span class="hl str">&quot;concordant&quot;</span><span class="hl std">,</span> <span class="hl num">NA</span><span class="hl std">,</span> <span class="hl str">&quot;discordant&quot;</span><span class="hl std">),</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">NA</span><span class="hl std">, COLc,</span> <span class="hl num">NA</span><span class="hl std">, COLd),</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl num">20</span><span class="hl std">,</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">title</span><span class="hl std">=</span><span class="hl str">&quot;SuRE vs eQTL&quot;</span><span class="hl std">)</span>
<span class="hl kwd">plot.new</span><span class="hl std">()</span> <span class="hl com">#fill up </span>
<span class="hl kwd">plot.new</span><span class="hl std">()</span> <span class="hl com">#fill up </span>
<span class="hl kwd">dev.off</span><span class="hl std">()</span>
</pre></div>

<div class="output"><pre class="knitr r">## png 
##   2
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">######################</span>
<span class="hl com">#zoom in on interesting region ABCC11:</span>
<span class="hl std">g</span><span class="hl kwb">&lt;-</span><span class="hl str">&quot;ABCC11&quot;</span>
<span class="hl std">win</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">48249000</span><span class="hl std">,</span> <span class="hl num">48251000</span><span class="hl std">)</span> <span class="hl com">#manually picked</span>
<span class="hl kwd">print</span><span class="hl std">(</span><span class="hl kwd">paste</span><span class="hl std">(g,</span> <span class="hl str">&quot;zoom&quot;</span><span class="hl std">))</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;ABCC11 zoom&quot;
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_SuRE_eQTL_overlap_HepG2_&quot;</span><span class="hl std">, g,</span> <span class="hl str">&quot;_zoom_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
       <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">6</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">3.5</span><span class="hl std">,</span> <span class="hl kwc">onefile</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
<span class="hl com">#layout parameters:</span>
<span class="hl kwd">layout</span><span class="hl std">(m)</span>
<span class="hl kwd">par</span><span class="hl std">(</span><span class="hl kwc">mar</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">mgp</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1.8</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0</span><span class="hl std">),</span> <span class="hl kwc">oma</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">4</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">,</span><span class="hl num">0.5</span><span class="hl std">),</span> <span class="hl kwc">lend</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">)</span>

<span class="hl std">Qls</span><span class="hl kwb">&lt;-</span><span class="hl std">Ql[Ql</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g</span> <span class="hl opt">&amp;</span> <span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl opt">&gt;=</span><span class="hl std">win[</span><span class="hl num">1</span><span class="hl std">]</span> <span class="hl opt">&amp;</span> <span class="hl std">Ql</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl opt">&lt;=</span><span class="hl std">win[</span><span class="hl num">2</span><span class="hl std">],]</span>
<span class="hl std">DQls</span><span class="hl kwb">&lt;-</span><span class="hl std">DQl[DQl</span><span class="hl opt">$</span><span class="hl std">gene_name</span><span class="hl opt">==</span><span class="hl std">g</span> <span class="hl opt">&amp;</span> <span class="hl std">DQl</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl opt">&gt;=</span><span class="hl std">win[</span><span class="hl num">1</span><span class="hl std">]</span> <span class="hl opt">&amp;</span> <span class="hl std">DQl</span><span class="hl opt">$</span><span class="hl std">snp_pos</span><span class="hl opt">&lt;=</span><span class="hl std">win[</span><span class="hl num">2</span><span class="hl std">],]</span>
<span class="hl com">#eGene start/end: compared to version above, this may lead to different x-axis range if outer SNPs do not have SuRE value</span>
<span class="hl std">gstart</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">orientation[</span><span class="hl num">1</span><span class="hl std">]</span><span class="hl opt">==</span><span class="hl str">&quot;+&quot;</span><span class="hl std">, DQls</span><span class="hl opt">$</span><span class="hl std">start[</span><span class="hl num">1</span><span class="hl std">], DQls</span><span class="hl opt">$</span><span class="hl std">end[</span><span class="hl num">1</span><span class="hl std">])</span>
<span class="hl std">gend</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">orientation[</span><span class="hl num">1</span><span class="hl std">]</span><span class="hl opt">==</span><span class="hl str">&quot;-&quot;</span><span class="hl std">, DQls</span><span class="hl opt">$</span><span class="hl std">start[</span><span class="hl num">1</span><span class="hl std">], DQls</span><span class="hl opt">$</span><span class="hl std">end[</span><span class="hl num">1</span><span class="hl std">])</span>
<span class="hl std">chr</span><span class="hl kwb">&lt;-</span><span class="hl kwd">unique</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">chrom); chr</span><span class="hl kwb">&lt;-</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;chr&quot;</span><span class="hl std">, chr)</span>
<span class="hl com">#SuRE SNP scores:</span>
<span class="hl std">rr</span><span class="hl kwb">&lt;-</span><span class="hl std">DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.ref.mean</span>
<span class="hl std">aa</span><span class="hl kwb">&lt;-</span><span class="hl std">DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.alt.mean</span>
<span class="hl std">sylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">15</span><span class="hl std">, aa[DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.wilcox.p.value</span><span class="hl opt">&lt;</span><span class="hl std">pplotcut],rr[DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.wilcox.p.value</span><span class="hl opt">&lt;</span><span class="hl std">pplotcut]),</span> <span class="hl kwc">na.rm</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">))</span> <span class="hl com">#SuRE y-axis range</span>

  <span class="hl com">#SuRE plot: </span>
   <span class="hl std">segcol</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ifelse</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">concordant, COLc, COLd)</span>
   <span class="hl kwd">plot</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">snp_pos, rr,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=sylim,</span> <span class="hl kwc">xaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">segments</span><span class="hl std">(</span><span class="hl kwc">x0</span><span class="hl std">=DQls</span><span class="hl opt">$</span><span class="hl std">snp_pos,</span> <span class="hl kwc">y0</span><span class="hl std">=rr,</span> <span class="hl kwc">y1</span><span class="hl std">=aa,</span> <span class="hl kwc">col</span><span class="hl std">=segcol,</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl opt">-</span><span class="hl kwd">log10</span><span class="hl std">(DQls</span><span class="hl opt">$</span><span class="hl std">hepg2.wilcox.p.value)</span><span class="hl opt">*</span><span class="hl std">p2width)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl str">&quot;SuRE signal&quot;</span><span class="hl std">,</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">2.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.8</span><span class="hl std">)</span>
   <span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topleft&quot;</span><span class="hl std">,</span> <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">paste</span><span class="hl std">(g,</span> <span class="hl str">&quot;(HepG2)&quot;</span><span class="hl std">),</span> <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">1.2</span><span class="hl std">)</span>
   <span class="hl kwd">legend</span><span class="hl std">(</span><span class="hl str">&quot;topright&quot;</span><span class="hl std">,</span>
          <span class="hl kwc">legend</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;coverage = &quot;</span><span class="hl std">,</span> <span class="hl kwd">nrow</span><span class="hl std">(DQls),</span> <span class="hl str">&quot;/&quot;</span><span class="hl std">,</span> <span class="hl kwd">nrow</span><span class="hl std">(Qls)),</span>
          <span class="hl kwc">bty</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>

   <span class="hl com">#eQTL scores:</span>
   <span class="hl std">eylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P))</span>
   <span class="hl kwd">plot</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">snp_pos, Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=eylim,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;h&quot;</span><span class="hl std">,</span> <span class="hl kwc">ylab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">xaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl str">&quot;eQTL&quot;</span><span class="hl std">,</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">2.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.8</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl kwd">expression</span><span class="hl std">(</span><span class="hl str">&quot;log&quot;</span><span class="hl std">[</span><span class="hl num">10</span><span class="hl std">]</span><span class="hl opt">*</span><span class="hl str">&quot;(P-value)&quot;</span><span class="hl std">),</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">2</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">1.5</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.7</span><span class="hl std">)</span>
   <span class="hl kwd">mtext</span><span class="hl std">(</span><span class="hl kwd">paste</span><span class="hl std">(</span><span class="hl str">&quot;position on&quot;</span><span class="hl std">,chr),</span> <span class="hl kwc">side</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl kwc">outer</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">,</span> <span class="hl kwc">line</span><span class="hl std">=</span><span class="hl num">1.8</span><span class="hl std">,</span> <span class="hl kwc">cex</span><span class="hl std">=</span><span class="hl num">0.7</span><span class="hl std">)</span>

   <span class="hl com">#room for Gviz annotation track:</span>
   <span class="hl std">aylim</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span> <span class="hl kwd">max</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P))</span>
   <span class="hl kwd">plot</span><span class="hl std">(Qls</span><span class="hl opt">$</span><span class="hl std">snp_pos, Qls</span><span class="hl opt">$</span><span class="hl std">MLOG10P,</span> <span class="hl kwc">xlim</span><span class="hl std">=win,</span> <span class="hl kwc">ylim</span><span class="hl std">=aylim,</span> <span class="hl kwc">xlab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">,</span> <span class="hl kwc">ylab</span><span class="hl std">=</span><span class="hl str">&quot;&quot;</span><span class="hl std">,</span> <span class="hl kwc">yaxt</span><span class="hl std">=</span><span class="hl str">&quot;n&quot;</span><span class="hl std">)</span>
   <span class="hl kwd">arrows</span><span class="hl std">(</span><span class="hl kwc">x0</span><span class="hl std">=gstart,</span> <span class="hl kwc">y0</span><span class="hl std">=aylim[</span><span class="hl num">2</span><span class="hl std">]</span><span class="hl opt">/</span><span class="hl num">10</span><span class="hl std">,</span> <span class="hl kwc">x1</span><span class="hl std">=gend,</span> <span class="hl kwc">length</span><span class="hl std">=</span><span class="hl num">0.05</span><span class="hl std">,</span> <span class="hl kwc">angle</span><span class="hl std">=</span><span class="hl num">30</span><span class="hl std">,</span> <span class="hl kwc">lwd</span><span class="hl std">=</span><span class="hl num">3</span><span class="hl std">,</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl str">&quot;darkgrey&quot;</span><span class="hl std">)</span> <span class="hl com">#plot eGene for later checks</span>
   <span class="hl std">xlim.full</span><span class="hl kwb">&lt;-</span><span class="hl kwd">round</span><span class="hl std">(</span><span class="hl kwd">par</span><span class="hl std">(</span><span class="hl str">&quot;usr&quot;</span><span class="hl std">)[</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">2</span><span class="hl std">)])</span> <span class="hl com">#actual full plotting window, to be passed on to gviz functions.</span>

<span class="hl kwd">dev.off</span><span class="hl std">()</span>
</pre></div>

<div class="output"><pre class="knitr r">## png 
##   2
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl com">##Now generate gene track using gviz:</span>
<span class="hl kwd">pdf</span><span class="hl std">(</span><span class="hl kwc">file</span><span class="hl std">=</span><span class="hl kwd">paste0</span><span class="hl std">(</span><span class="hl str">&quot;FigurePanel_Gviz_annotation_HepG2_&quot;</span><span class="hl std">, g,</span> <span class="hl str">&quot;_zoom_BvS&quot;</span><span class="hl std">,Date,</span> <span class="hl str">&quot;.pdf&quot;</span><span class="hl std">),</span>
       <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl num">12.2</span><span class="hl std">,</span> <span class="hl kwc">height</span><span class="hl std">=</span><span class="hl num">1.1</span><span class="hl std">)</span> <span class="hl com">#broad for full locus</span>

   <span class="hl std">biomTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">BiomartGeneRegionTrack</span><span class="hl std">(</span><span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
                                    <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
                                    <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
                                    <span class="hl kwc">collapseTranscripts</span><span class="hl std">=</span><span class="hl str">&quot;meta&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">transcriptAnnotation</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span>
                                    <span class="hl kwc">cex.group</span><span class="hl std">=</span><span class="hl num">0.5</span><span class="hl std">,</span>
                                    <span class="hl kwc">shape</span><span class="hl std">=</span><span class="hl str">&quot;smallArrow&quot;</span><span class="hl std">,</span>
                                    <span class="hl kwc">col.line</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span> <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl kwa">NULL</span><span class="hl std">,</span> <span class="hl kwc">fill</span><span class="hl std">=</span><span class="hl str">&quot;darkred&quot;</span><span class="hl std">,</span>
                                    <span class="hl com">#background.panel = &quot;#EEEEEE&quot;,</span>
                                    <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;Genes&quot;</span><span class="hl std">)</span>

  <span class="hl std">GR</span><span class="hl kwb">&lt;-</span><span class="hl kwd">makeGRangesFromDataFrame</span><span class="hl std">(</span><span class="hl kwd">data.frame</span><span class="hl std">(</span><span class="hl kwc">chr</span><span class="hl std">=chr,</span> <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span> <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span> <span class="hl kwc">score</span><span class="hl std">=</span><span class="hl num">1</span><span class="hl std">))</span>
  <span class="hl com">#DNase track:</span>
  <span class="hl std">DNase</span><span class="hl kwb">&lt;-</span><span class="hl kwd">import</span><span class="hl std">(URL.DNase.HepG2,</span> <span class="hl kwc">which</span><span class="hl std">=GR)</span> <span class="hl com">#grab only relevant region </span>
  <span class="hl std">dnaseTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">DataTrack</span><span class="hl std">(</span><span class="hl kwc">range</span><span class="hl std">=DNase,</span>
                            <span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
                            <span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
                            <span class="hl kwc">end</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
                            <span class="hl kwc">type</span><span class="hl std">=</span><span class="hl str">&quot;h&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">ylim</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">0</span><span class="hl std">,</span><span class="hl num">200</span><span class="hl std">),</span>
                            <span class="hl kwc">col</span><span class="hl std">=</span><span class="hl str">&quot;darkgreen&quot;</span><span class="hl std">,</span>
                            <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;DNaseI&quot;</span><span class="hl std">)</span> <span class="hl com">#scale adjusted to lower dynamic range compared to K562</span>

  <span class="hl com">#create simple full-width bar to facilatate alignment in bottom frame of SuRE/eQTL plot:</span>
  <span class="hl std">aTrack</span><span class="hl kwb">&lt;-</span><span class="hl kwd">AnnotationTrack</span><span class="hl std">(</span><span class="hl kwc">start</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span> <span class="hl kwc">width</span><span class="hl std">=</span><span class="hl kwd">diff</span><span class="hl std">(xlim.full),</span> <span class="hl kwc">strand</span><span class="hl std">=</span><span class="hl str">&quot;*&quot;</span><span class="hl std">,</span> <span class="hl kwc">genome</span><span class="hl std">=</span><span class="hl str">&quot;hg19&quot;</span><span class="hl std">,</span> <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span> <span class="hl kwc">name</span><span class="hl std">=</span><span class="hl str">&quot;aln&quot;</span><span class="hl std">)</span>

  <span class="hl kwd">plotTracks</span><span class="hl std">(</span><span class="hl kwd">list</span><span class="hl std">(biomTrack, dnaseTrack, aTrack),</span>
             <span class="hl kwc">chromosome</span><span class="hl std">=chr,</span>
             <span class="hl kwc">from</span><span class="hl std">=xlim.full[</span><span class="hl num">1</span><span class="hl std">],</span>
             <span class="hl kwc">to</span><span class="hl std">=xlim.full[</span><span class="hl num">2</span><span class="hl std">],</span>
             <span class="hl kwc">sizes</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl std">,</span><span class="hl num">1</span><span class="hl std">,</span> <span class="hl num">0.3</span><span class="hl std">),</span>
             <span class="hl kwc">fontcolor.title</span><span class="hl std">=</span><span class="hl str">&quot;black&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">col.border.title</span><span class="hl std">=</span><span class="hl str">&quot;transparent&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">background.title</span><span class="hl std">=</span><span class="hl str">&quot;transparent&quot;</span><span class="hl std">,</span>
             <span class="hl kwc">col.axis</span><span class="hl std">=</span><span class="hl str">&quot;black&quot;</span>
             <span class="hl std">)</span>


<span class="hl kwd">dev.off</span><span class="hl std">()</span> <span class="hl com">#close Gviz PDF</span>
</pre></div>

<div class="output"><pre class="knitr r">## png 
##   2
</pre></div>

<p></div></div></p>

<p><b>Gviz plots can now be pasted into bottom frame of SuRE/eQTL plots using Illustrator. The width of the former should be adjusted until the light-blue bar fits <i>exactly</i> in the frame. As an additional check, then the eGene according to Gviz should overlap perfectly with the grey arrow. The grey arrow and the light-blue bar may then be removed manually.</b> </p>

<h3>Session Info</h3>

<div class="chunk" id="html-SessionInfo"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl kwd">sessionInfo</span><span class="hl std">()</span>
</pre></div>

<div class="output"><pre class="knitr r">## R version 3.4.4 (2018-03-15)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.5 LTS
## 
## Matrix products: default
## BLAS: /usr/lib/libblas/libblas.so.3.6.0
## LAPACK: /usr/lib/lapack/liblapack.so.3.6.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
## [1] Gviz_1.20.0          rtracklayer_1.38.3   GenomicRanges_1.30.3
## [4] GenomeInfoDb_1.14.0  IRanges_2.12.0       S4Vectors_0.16.0    
## [7] BiocGenerics_0.24.0  knitr_1.20           data.table_1.11.4   
## 
## loaded via a namespace (and not attached):
##  [1] Biobase_2.38.0                httr_1.3.1                   
##  [3] bit64_0.9-7                   AnnotationHub_2.8.3          
##  [5] splines_3.4.4                 shiny_1.1.0                  
##  [7] Formula_1.2-3                 interactiveDisplayBase_1.14.0
##  [9] highr_0.6                     latticeExtra_0.6-28          
## [11] blob_1.1.1                    BSgenome_1.46.0              
## [13] GenomeInfoDbData_1.0.0        Rsamtools_1.30.0             
## [15] yaml_2.1.19                   pillar_1.2.2                 
## [17] RSQLite_2.1.1                 backports_1.1.2              
## [19] lattice_0.20-35               biovizBase_1.24.0            
## [21] digest_0.6.15                 promises_1.0.1               
## [23] RColorBrewer_1.1-2            XVector_0.18.0               
## [25] checkmate_1.8.5               colorspace_1.3-2             
## [27] httpuv_1.4.3                  htmltools_0.3.6              
## [29] Matrix_1.2-14                 plyr_1.8.4                   
## [31] XML_3.98-1.11                 biomaRt_2.32.1               
## [33] zlibbioc_1.24.0               xtable_1.8-2                 
## [35] scales_0.5.0                  later_0.7.2                  
## [37] BiocParallel_1.12.0           htmlTable_1.12               
## [39] tibble_1.4.2                  AnnotationFilter_1.0.0       
## [41] ggplot2_2.2.1                 SummarizedExperiment_1.8.1   
## [43] GenomicFeatures_1.28.5        nnet_7.3-12                  
## [45] lazyeval_0.2.1                mime_0.5                     
## [47] survival_2.42-6               magrittr_1.5                 
## [49] evaluate_0.10.1               memoise_1.1.0                
## [51] foreign_0.8-70                BiocInstaller_1.26.1         
## [53] tools_3.4.4                   matrixStats_0.53.1           
## [55] stringr_1.3.1                 munsell_0.4.3                
## [57] cluster_2.0.7-1               DelayedArray_0.4.1           
## [59] AnnotationDbi_1.38.2          ensembldb_2.0.4              
## [61] Biostrings_2.46.0             compiler_3.4.4               
## [63] rlang_0.2.0                   RCurl_1.95-4.10              
## [65] dichromat_2.0-0               rstudioapi_0.7               
## [67] VariantAnnotation_1.22.3      htmlwidgets_1.2              
## [69] bitops_1.0-6                  base64enc_0.1-3              
## [71] gtable_0.2.0                  curl_3.2                     
## [73] DBI_1.0.0                     R6_2.2.2                     
## [75] GenomicAlignments_1.14.2      gridExtra_2.3                
## [77] bit_1.1-14                    Hmisc_4.1-1                  
## [79] ProtGenerics_1.8.0            stringi_1.2.2                
## [81] Rcpp_0.12.16                  rpart_4.1-13                 
## [83] acepack_1.4.1
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl kwd">getwd</span><span class="hl std">()</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;/DATA/usr/b.v.steensel/Projects/Git/SuRE/Bas/SNP/eQTL&quot;
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl kwd">date</span><span class="hl std">()</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;Tue Oct  2 10:12:03 2018&quot;
</pre></div>

<div class="source"><pre class="knitr r"><span class="hl kwd">paste</span><span class="hl std">(</span><span class="hl str">&quot;Run time: &quot;</span><span class="hl std">,</span><span class="hl kwd">format</span><span class="hl std">(</span><span class="hl kwd">Sys.time</span><span class="hl std">()</span><span class="hl opt">-</span><span class="hl std">StartTime))</span>
</pre></div>

<div class="output"><pre class="knitr r">## [1] &quot;Run time:  3.29641 mins&quot;
</pre></div>

<p></div></div></p>

<p></body>
</html></p>

</body>

</html>
